app/models/smart_proxy.rb

```diff
@@ -88,22 +86,25 @@ def has_feature?(feature)
     self.features.any? { |proxy_feature| proxy_feature.name == feature }
   end
 
+  def features_to_lowercase
+    features.pluck("LOWER(REPLACE(name, ' ', '_'))").uniq
+  end
+
   def version
-    result = {}
-    begin
-      Timeout::timeout(20) do
-        version = ProxyAPI::Version.new(:url => url).version
-        result[:success] = true
-        result[:message] = version
-      end
-    rescue *HTTP_ERRORS => exception
-      raise ::Foreman::WrappedException.new exception, N_("Unable to connect to smart proxy")
-    end
-    result
+    proxy_status.proxy_versions
```

Comment:
`delegate :version, :tftp_server, :to => :proxy_status`, handle the 'no tftp feature' there directly.